subsys/bluetooth/mesh/main.c

```diff
@@ -169,7 +167,7 @@ int bt_mesh_provision(const uint8_t net_key[16], uint16_t net_idx,
 	bt_mesh_start();
 
 end:
-	if (err && is_cdb_dev_key_valid && IS_ENABLED(CONFIG_BT_MESH_CDB)) {
+	if (err && node != NULL) {
```

Comment:
```suggestion
	if (IS_ENABLED(CONFIG_BT_MESH_CDB) && err && node != NULL) {
```
`bt_mesh_cdb_node_del` is only compiled if `CONFIG_BT_MESH_CDB` is enabled.

Reply:
The code will be compiled out anyway, as `node` is always NULL if `CONFIG_BT_MESH_CDB` is disabled

Reply:
When compiling with `CONFIG_NO_OPTIMIZATIONS`:

> [203/203] Linking C executable zephyr/zephyr.elf
...
/usr/bin/ld.bfd: app/libapp.a(main.c.obj): warning: relocation in read-only section `.rodata.comp'
/usr/bin/ld.bfd: zephyr/subsys/bluetooth/mesh/libsubsys__bluetooth__mesh.a(main.c.obj): in function `bt_mesh_provision':
/home/pavel/nordic/ncs/git-repos/zephyrproject/zephyr/subsys/bluetooth/mesh/main.c:171: undefined reference to `bt_mesh_cdb_node_del'

Reply:
done